src/main/scala/com/codacy/parsers/implementation/CoberturaParser.scala

```diff
@@ -3,41 +3,34 @@ package com.codacy.parsers.implementation
 import java.io.File
 
 import com.codacy.api.{CoverageFileReport, CoverageReport}
-import com.codacy.parsers.CoverageParser
-import com.codacy.parsers.util.{TextUtils, XMLoader}
+import com.codacy.parsers.util.TextUtils
+import com.codacy.parsers.{CoverageParser, XmlReportParser}
 
-import scala.util.{Failure, Success, Try}
 import scala.xml.{Elem, NodeSeq}
 
-object CoberturaParser extends CoverageParser {
+object CoberturaParser extends CoverageParser with XmlReportParser {
 
   override val name: String = "Cobertura"
 
-  def parse(projectRoot: File, reportFile: File): Either[String, CoverageReport] = {
-    val report = (Try(XMLoader.loadFile(reportFile)) match {
-      case Success(xml) if hasCorrectSchema(xml) =>
-        Right(xml \\ "coverage")
-
-      case Success(_) =>
-        Left("Invalid report. Could not find top level <coverage> tag.")
+  private val CoverageTag = "coverage"
+  private val LineRateAttribute = "@line-rate"
```

Comment:
I would not put the `@` inside the `val` but then use string concatenation like:
```scala
private val LineRate = "line-rate"
a \ b \ s"@$LineRate"
```

Reply:
But then, I would have to use the s"@$LineRate in three different places. It seems weird. Why do you think that would be better?

Reply:
Nevermind, If I change this, I will just use \@ in the remaining usages

Reply:
Done ✅